ractive-koa/app.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+// - - - - - - - - - - - - - - - - - - - - - - - - >>>>> vars
+
+var ss = require('socketstream');
+var ssJade = require('ss-jade');
+var ssStylus = require('ss-stylus');
+var koa = require('koa');
+var connect = require('koa-connect');
+var session = require('koa-session');
+var router = require('koa-router')();
+var app = koa();
+var server;
+
+// - - - - - - - - - - - - - - - - - - - - - - - - >>>>> config
+
+var config = require('./server/services/config');
+// var dbService = require('./server/services/db');
+
+// - - - - - - - - - - - - - - - - - - - - - - - - >>>>> db services
+
+// dbService.connectMongoose();
+// dbService.connectRedis();
+
+// - - - - - - - - - - - - - - - - - - - - - - - - >>>>> HTML FORMATTER
+
+// if (ss.env === 'development') {
+// 	ssJade.addCompileOptions({
+// 		pretty: true
+// 	});
+// }
+ss.client.formatters.add(ssJade, {
+	locals: {
+		// title: 'Arx :: Home',
+		// mobileAgents: JSON.stringify(require(rootDir + '/data/mobile-agents')),
+		// SubAtomic: JSON.stringify(SubAtomic.getGlobals()),
+		// mobile: false
+	},
+});
+
+// - - - - - - - - - - - - - - - - - - - - - - - - >>>>> STYLE FORMATTER
+
+
+// ssStylus.prependStylus(
+// 	'$cloudPath = \'' + cloudPath + '/\'\n' +
+// 	'$cache = \'' + SubAtomic.get('CACHE') + '\''
+// );
+ss.client.formatters.add(ssStylus);
+
+// - - - - - - - - - - - - - - - - - - - - - - - - >>>>> TEMPLATE ENGINE
+
+ss.client.templateEngine.use(require('ss-ractive'), '/', {
+	pretty: ss.env === 'production' ? false : true
+});
+
+// - - - - - - - - - - - - - - - - - - - - - - - - >>>>> ROUTING & KOA
+
+app.keys = [config.get('session:secret')];
+app.use(session(app));
+app.use(router.routes());
+app.use(router.allowedMethods());
+
+// Append SocketStream middleware to the stack
+require('./server/services/routes')(ss, app, router);
+app.use(connect(ss.http.middleware));
+
+// - - - - - - - - - - - - - - - - - - - - - - - - >>>>> PACK ASSETS
+
+// Minimize and pack assets if you type: SS_ENV=production node app.js
+if (ss.env === 'production') {
+	ss.client.packAssets();
+}
+
+// server = http.createServer(app.callback()).listen(config.get('port'), function() {
+// 	console.log(server.address());
+// 	ss.start(server);
+// });
+
+
+server = app.listen(config.get('port'), function() {
```

Comment:
With the middleware mounted, I use Koa's `app.listen` and then pass the `server` var to `ss.start` below.